### ISSUE TITLE
complete moving build agents to DO

### ISSUE THREAD

**Issue by benesch** [cockroachdb/cockroach#19270]
Digital Ocean agents seem to have slower CPUs than our Azure agents, which exacerbates the overhead of the race detector. This has turned up several new flaky tests:

  - [x] `storage.TestScanStoreIntents` (#19266)
  -   [x] `sql.TestAsOfRetry` (#19268)
  -   [x] `server.TestRangeResponse` and `server.TestRangesResponse` (#19264)
  -   [ ] `sql/logictest.TestLogic` (!) (#19269)
  -   [ ] `TestReplicateQueue...` (#19254, #19249, #19250)

So far, every single one of these failures has been a legitimate bug in the test.

Also worth noting: the 12 CPU droplet _consistently_ fails `kv.TestLeaseHolderCache` under race. I'll bet that's a real bug, too.

Once these bugs are all fixed we can turn the Digital Ocean build agents back on. Back on Azure for now.

**Comment by benesch** [comment]
tl;dr we sidestepped most of these issues by using High CPU instances. 🎉
